src/arduino.cc/builder/builder_utils/utils.go

```diff
@@ -258,6 +289,24 @@ func nonEmptyString(s string) bool {
 	return s != constants.EMPTY_STRING
 }
 
+func CheckIfRecompileIsAvoidable(corePath string, archiveFile string) bool {
```

Comment:
Correct, the timestamps are only extracted from `core` subfolder, while they should be applied to the whole folder. I'm pushing another commit to fix this